js/notifications.js

```diff
@@ -12,38 +12,27 @@
         MESSAGE : 'message'
     };
 
+    var enabled = false;
+
     Whisper.Notifications = new (Backbone.Collection.extend({
         initialize: function() {
+            this.on('add', this.update);
             this.on('remove', this.onRemove);
         },
-
-        onRemove: function() {
-            console.log('remove notification');
-        },
-        clear: function() {
-            console.log('remove all notifications');
-            this.reset([]);
-        },
-
-        getSetting: function() {
-            return storage.get('notification-setting') || SETTINGS.MESSAGE;
-        },
         onClick: function(conversationId) {
             var conversation = ConversationController.get(conversationId);
             this.trigger('click', conversation);
         },
         update: function() {
             console.log(
-                'updating notifications - count:',
-                this.length,
-                'focused:',
-                window.isFocused()
+                'updating notifications - count:', this.length,
+                'focused:', window.isFocused(),
+                'enabled:', enabled
             );
-
-            if (this.length === 0) {
-                return;
+            if (!enabled) {
+                return; // wait til we are re-enabled
             }
-            if (window.isFocused()) {
```

Comment:
Without this check, we'll show notifications when they come in after the 'empty' event, even if the window is focused. Or am I missing something?

Reply:
Hmm.. looks like there's a focus check in conversation.notify that accounts for that case, but it probably makes sense to move that check here. Will revise.